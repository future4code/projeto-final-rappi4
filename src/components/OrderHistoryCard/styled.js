import styled from 'styled-components';

export const StyledCardContainer = styled.div`
    height: 112.6px;
    border-radius: 8px;
    border: solid 1px #b8b8b8;
    padding: 16px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`

export const StyledTitleSpan = styled.span`
    display: block;
    color: #e86e5a;
    font-size: 16px;
`
export const StyledContentSpan = styled.span`
    display: block;
    font-size: 12px;
`
export const StyledPriceSpan = styled.span`
    display: block;
    font-size: 16px;
    font-weight: bold;
`
