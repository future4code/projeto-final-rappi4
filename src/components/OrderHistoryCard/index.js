import React from 'react'
import { StyledCardContainer, StyledTitleSpan, StyledContentSpan, StyledPriceSpan, StyledCardContent } from './styled';


export default function OrderHistoryCard(props) {
    const orderDate = new Date(props.order.createdAt).toLocaleDateString();
    return (
        <StyledCardContainer>
            <StyledTitleSpan>{props.order.restaurantName}</StyledTitleSpan>
            <StyledContentSpan>{orderDate}</StyledContentSpan>
            <StyledPriceSpan>SUBTOTAL R$ {props.order.totalPrice},00</StyledPriceSpan>
        </StyledCardContainer>
    )
}