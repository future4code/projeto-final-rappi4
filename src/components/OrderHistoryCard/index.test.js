import React from 'react';
import renderer from 'react-test-renderer';
import OrderHistoryCard from './index';


describe("OrderHistoryCard",()=>{
    it("should render",()=>{
        const mockOrder = {
            totalPrice: 20,
            restaurantName: "Habibs",
            createdAt: 1574659218053,
            expiresAt: 1574662818053,
        }
        const tree = renderer.create(<OrderHistoryCard order={mockOrder}/>).toJSON();

        expect(tree).toMatchSnapshot();
    })
})