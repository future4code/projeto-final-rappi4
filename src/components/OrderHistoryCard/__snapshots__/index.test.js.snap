// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderHistoryCard should render 1`] = `
<div
  className="sc-bdVaJa euQVtR"
>
  <span
    className="sc-bwzfXH bDFBMC"
  >
    Habibs
  </span>
  <span
    className="sc-htpNat gSoASV"
  >
    2019-11-25
  </span>
  <span
    className="sc-bxivhb fVumoj"
  >
    SUBTOTAL R$ 
    20
    ,00
  </span>
</div>
`;
